### PR TITLE
add fulltextsearch as option

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/issues/1036

- [x] needs https://github.com/nextcloud-releases/all-in-one/pull/28

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: Containers/nextcloud/entrypoint.sh [hunk 1]
```diff
@@ -423,5 +423,47 @@ if version_greater "24.0.0.0" "$installed_version"; then
     fi
 fi
 
+# Fulltextsearch
+if [ "$FULLTEXTSEARCH_ENABLED" = 'yes' ]; then
+    while ! nc -z "$FULLTEXTSEARCH_HOST" 9200; do
+        echo "waiting for OnlyOffice to become available..."
```
Comment (Zoey2936) [inline Containers/nextcloud/entrypoint.sh:429]: I think this is not OnlyOffice

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: fixed

File: Containers/nextcloud/activate-fulltextsearch.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+if [ "$FULLTEXTSEARCH_ENABLED" != yes ]; then
+    # Basically sleep for forever if fulltextsearch is not enabled
+    sleep 365d
+fi
+echo "Activating fulltextsearch..."
+php /var/www/html/occ fulltextsearch:live -q
+sleep 365d
```
Comment (Zoey2936) [inline Containers/nextcloud/activate-fulltextsearch.sh:9]:
```suggestion
    sleep inf
fi
echo "Activating fulltextsearch..."
php /var/www/html/occ fulltextsearch:live -q
sleep inf
```
Why not directly sleep forever?
See: https://www.gnu.org/software/coreutils/manual/html_node/sleep-invocation.html

Comment (szaimen) [reply Containers/nextcloud/activate-fulltextsearch.sh:9]: good point! didn't know that this is possible...

Comment (szaimen) [reply Containers/nextcloud/activate-fulltextsearch.sh]: fixed

File: Containers/fulltextsearch/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Probably from here https://github.com/elastic/elasticsearch/blob/main/distribution/docker/src/docker/Dockerfile
+FROM elasticsearch:7.17.5
+
+RUN elasticsearch /usr/share/elasticsearch/bin/elasticsearch-plugin install --batch ingest-attachment
```
Comment (Zoey2936) [inline Containers/fulltextsearch/Dockerfile:4]: I think this does not build
grafik

If I replace the RUN command with: `elasticsearch-plugin install --batch ingest-attachment`
It outputs:
grafik

Comment (szaimen) [reply Containers/fulltextsearch/Dockerfile:4]: you are right. Fixed with https://github.com/nextcloud/all-in-one/commit/6c63adc37ff0f755375cea2a518bb0adc11560f4

**Comment by szaimen** [comment]
@Zoey2936 This is the last remaning thing that I was not yet able to confirm that it works from [this list](https://github.com/nextcloud/all-in-one/issues/1041). Did you tried it out already? I am not sure what I am currently missing why it doesn't work...

**Comment by Zoey2936** [comment]
> @Zoey2936 This is the last remaning thing that I was not yet able to confirm that it works from [this list](https://github.com/nextcloud/all-in-one/issues/1041). Did you tried it out already? I am not sure what I am currently missing why it doesn't work...

I tried but it seems that it does not work
Running `docker exec -it nextcloud-aio-nextcloud php occ fulltextsearch:index` did something, but I don't know what exactly

**Comment by Zoey2936** [comment]
grafik

**Comment by szaimen** [comment]
Thanks for checking! I've also asked a colleague that has more knowledge regarding this. Let's see.

**Comment by Zoey2936** [comment]
grafik
Now it reports this error

**Comment by szaimen** [comment]
Yeah, seems like this is not really stable and usable currently... Will see on monday if I am able to fix this with the help of a colleague...

**Comment by szaimen** [comment]
Fixed with https://github.com/nextcloud/all-in-one/pull/1065

**Pull request merged**